packages/enzyme/src/ShallowWrapper.js

```diff
@@ -339,6 +339,9 @@ class ShallowWrapper {
             && typeof instance.shouldComponentUpdate === 'function'
           ) {
             spy = spyMethod(instance, 'shouldComponentUpdate');
+          // PureComponent
+          } else if (instance.isPureReactComponent) {
+            shouldRender = !isEqual(prevProps, props) || !isEqual(state, instance.state);
```

Comment:
let's put this in some sort of `pureComponentShouldComponentUpdate` function, so here and line 471 can share the implementation.